examples/image_segmentation/ios/ImageSegmenter/ViewContoller/MediaLibraryViewController.swift

```diff
@@ -0,0 +1,384 @@
+                let imageSegmenterResult = resultBundle.imageSegmenterResults.first,
+                let imageSegmenterResult = imageSegmenterResult else {
+            DispatchQueue.main.async {
+              self?.hideProgressView()
+            }
+            return
+          }
+
+          DispatchQueue.main.async {
+            self.hideProgressView()
+            self.render.prepare(with: image.size, outputRetainedBufferCountHint: 3)
+            self.inferenceResultDeliveryDelegate?.didPerformInference(result: resultBundle)
+            let marks = imageSegmenterResult.confidenceMasks
+            let _mark = marks![0]
+            let float32Data = _mark.float32Data
+            let newImage = self.render.render(image: image, segmentDatas: float32Data)
+            self.pickedImageView.image = newImage
+          }
+        }
+      default:
+        break
+      }
+    }
+
+  private func imageGenerator(with videoAsset: AVAsset) -> AVAssetImageGenerator {
+    let generator = AVAssetImageGenerator(asset: videoAsset)
+    generator.requestedTimeToleranceBefore = CMTimeMake(value: 1, timescale: 25)
+    generator.requestedTimeToleranceAfter = CMTimeMake(value: 1, timescale: 25)
+    generator.appliesPreferredTrackTransform = true
+
+    return generator
+  }
+
+  func clearAndInitializeImageSegmenterService(runningMode: RunningMode) {
+    imageSegmenterService = nil
+    switch runningMode {
+    case .image:
+      imageSegmenterService = ImageSegmenterService.stillImageSegmenterService(
+        modelPath: InferenceConfigurationManager.sharedInstance.modelPath)
+    case .video:
+      imageSegmenterService = ImageSegmenterService.videoImageSegmenterService(
+        modelPath: InferenceConfigurationManager.sharedInstance.modelPath)
+    default:
+      break;
+    }
+  }
+
+  func getVideoFormatDescription(from asset: AVAsset) -> (description: CMFormatDescription?, needChangeWidthHeight: Bool)  {
+    // Get the video track from the asset
+    guard let videoTrack = asset.tracks(withMediaType: AVMediaType.video).first else {
+      print("No video track found in the asset.")
+      return (nil, false)
+    }
+    let naturalSize = videoTrack.naturalSize
+    let size = videoTrack.naturalSize.applying(videoTrack.preferredTransform)
+    let newSize = CGSize(width: abs(size.width), height: abs(size.height))
+    var needChangeWidthHeight = false
+    if naturalSize.width == newSize.height {
+      needChangeWidthHeight = true
+    }
+
+    // Create an asset reader
+    do {
+      let assetReader = try AVAssetReader(asset: asset)
+
+      // Create an asset reader track output
+      let outputSettings: [String: Any] = [kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32BGRA]
+      let trackOutput = AVAssetReaderTrackOutput(track: videoTrack, outputSettings: outputSettings)
+      assetReader.add(trackOutput)
+
+      // Start the asset reader
+      if assetReader.startReading() {
```

Comment:
Convert the nested `if`s in this block to `guard let ` statements to improve readability